TestingLowerBounds/StatInfoFun.lean

```diff
@@ -340,12 +339,137 @@ end statInfoFun_γ
 section fDiv
 
 lemma nnReal_mul_fDiv {a : NNReal} :
-    a * fDiv (fun x ↦ statInfoFun β γ x) μ ν = fDiv (statInfoFun (a * β) (a * γ)) μ ν := by
+    a * fDiv (statInfoFun β γ) μ ν = fDiv (fun x ↦ statInfoFun (a * β) (a * γ) x) μ ν := by
   change (a.1 : EReal) * _ = _
   rw [← fDiv_mul a.2 ((convexOn_statInfoFun β γ).subset (fun _ _ ↦ trivial) (convex_Ici 0)) μ ν]
   simp_rw [const_mul_statInfoFun a.2]
   rfl
 
+lemma fDiv_statInfoFun_eq_integral_max_of_nonneg_of_le [IsFiniteMeasure μ] [IsFiniteMeasure ν]
+    (hβ : 0 ≤ β) (hγ : γ ≤ β) :
+    fDiv (statInfoFun β γ) μ ν = ∫ x, max 0 (γ - β * ((∂μ/∂ν) x).toReal) ∂ν := by
+  simp_rw [fDiv_of_integrable (integrable_statInfoFun_rnDeriv _ _ _ _),
+    derivAtTop_statInfoFun_of_nonneg_of_le hβ hγ, zero_mul, add_zero, statInfoFun_of_le hγ]
+
+lemma fDiv_statInfoFun_eq_integral_max_of_nonneg_of_gt [IsFiniteMeasure μ] [IsFiniteMeasure ν]
+    (hβ : 0 ≤ β) (hγ : β < γ) :
+    fDiv (statInfoFun β γ) μ ν
+      = ∫ x, max 0 (β * ((∂μ/∂ν) x).toReal - γ) ∂ν + β * (μ.singularPart ν) univ := by
+  simp_rw [fDiv_of_integrable (integrable_statInfoFun_rnDeriv _ _ _ _),
+    derivAtTop_statInfoFun_of_nonneg_of_gt hβ hγ, statInfoFun_of_gt hγ]
+
+lemma fDiv_statInfoFun_eq_integral_max_of_nonpos_of_le [IsFiniteMeasure μ] [IsFiniteMeasure ν]
+    (hβ : β ≤ 0) (hγ : γ ≤ β) :
+    fDiv (statInfoFun β γ) μ ν
+      = ∫ x, max 0 (γ - β * ((∂μ/∂ν) x).toReal) ∂ν - β * (μ.singularPart ν) univ := by
+  simp_rw [fDiv_of_integrable (integrable_statInfoFun_rnDeriv _ _ _ _),
+    derivAtTop_statInfoFun_of_nonpos_of_le hβ hγ, statInfoFun_of_le hγ, neg_mul, ← sub_eq_add_neg]
+
+lemma fDiv_statInfoFun_eq_integral_max_of_nonpos_of_gt [IsFiniteMeasure μ] [IsFiniteMeasure ν]
+    (hβ : β ≤ 0) (hγ : β < γ) :
+    fDiv (statInfoFun β γ) μ ν = ∫ x, max 0 (β * ((∂μ/∂ν) x).toReal - γ) ∂ν := by
+  simp_rw [fDiv_of_integrable (integrable_statInfoFun_rnDeriv _ _ _ _),
+    derivAtTop_statInfoFun_of_nonpos_of_gt hβ hγ, statInfoFun_of_gt hγ, zero_mul, add_zero]
+
+/-- Auxiliary lemma for `fDiv_statInfoFun_eq_integral_abs_of_nonneg_of_le` and
+`fDiv_statInfoFun_eq_integral_abs_of_nonpos_of_le`. -/
+lemma integral_max_eq_integral_abs [IsFiniteMeasure μ] [IsFiniteMeasure ν] :
+    ∫ x, max 0 (γ - β * ((∂μ/∂ν) x).toReal) ∂ν = (2 : ℝ)⁻¹ * (∫ x, |β * ((∂μ/∂ν) x).toReal
+      - γ| ∂ν + γ * (ν univ).toReal - β * (μ univ).toReal
+      + β * ((μ.singularPart ν) univ).toReal) := by
```

Comment:
Please don't go to a new line in the middle of an integral. Try to find better positions for the line breaks.